### PR TITLE
Home screen

### PR DESCRIPTION
**Pull request by hoangtuyb96**

### CODE REVIEW

File: app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenreFragment.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import android.widget.TextView;
+
+import com.example.admin.soundcloud_13.R;
+import com.example.admin.soundcloud_13.constant.Constant;
+import com.example.admin.soundcloud_13.screen.BaseFragment;
+
+public class GenreFragment extends BaseFragment {
+    private String mGenre;
+    private String mTitle;
+    private TextView textViewTitleGenre;
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenreFragment.java]: convention

File: app/src/main/java/com/example/admin/soundcloud_13/screen/Home/HomeActivity.java [hunk 2]
```diff
@@ -1,15 +1,43 @@
...
+    private HomeContract.Presenter mPresenter;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
+
+        TabLayout tabLayout = findViewById(R.id.tab_layout_home);
+        tabLayout.addTab(tabLayout.newTab().setText("Your Library"));
+        tabLayout.addTab(tabLayout.newTab().setText("Online"));
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/Home/HomeActivity.java]: 19 20 hardcode

File: app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenrePagerAdapter.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package com.example.admin.soundcloud_13.screen.Home.Online;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+public class GenrePagerAdapter extends FragmentPagerAdapter {
+    int tabCount;
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenrePagerAdapter.java]: no access  modifier? convention

File: app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenrePagerAdapter.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            case 0:
+                return GenreFragment.newInstance("all-audio");
+            case 1:
+                return GenreFragment.newInstance("alternativerock");
+            case 2:
+                return GenreFragment.newInstance("ambient");
+            case 3:
+                return GenreFragment.newInstance("classical");
+            case 4:
+                return GenreFragment.newInstance("country");
```
Comment (tuannguyen38) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/Home/Online/GenrePagerAdapter.java]: 18 -> 27 hardcode

File: app/src/main/java/com/example/admin/soundcloud_13/constant/Constant.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.example.admin.soundcloud_13.constant;
+
+public class Constant {
+    public final static String GENRE_KEY = "genre";
+    public final static String ALL_AUDIO = "all-audio";
+    public final static String ALTERNATIVE_ROCK = "alternativerock";
+    public final static String AMBIENT = "ambient";
+    public final static String CLASSICAL = "classical";
+    public final static String COUNTRY = "country";
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/soundcloud_13/constant/Constant.java]: https://developer.android.com/reference/android/support/annotation/StringDef
sử dụng string def

File: app/src/main/java/com/example/admin/soundcloud_13/constant/Constant.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package com.example.admin.soundcloud_13.constant;
+
+public class Constant {
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/soundcloud_13/constant/Constant.java]: rename to Constants

File: app/src/main/java/com/example/admin/soundcloud_13/screen/home/HomeActivity.java [hunk 7]
```diff
@@ -1,15 +1,42 @@
...
+
+        TabLayout tabLayout = findViewById(R.id.tab_layout_home);
+        tabLayout.addTab(tabLayout.newTab().setText(Constant.YOUR_LIBRARY));
+        tabLayout.addTab(tabLayout.newTab().setText(Constant.ONLINE));
+
+        final ViewPager viewPager = findViewById(R.id.view_pager_home);
+        final PagerAdapter pagerAdapter = new PagerAdapter(getSupportFragmentManager(), tabLayout.getTabCount());
+        viewPager.setAdapter(pagerAdapter);
+
+        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/home/HomeActivity.java]: sử dụng tablayout.setupwithViewpager

File: app/src/main/java/com/example/admin/soundcloud_13/screen/home/PagerAdapter.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+    public PagerAdapter(FragmentManager fm, int tabCount) {
+        super(fm);
+        this.mTabCount = tabCount;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position) {
+            case 0:
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/home/PagerAdapter.java]: nên sử dụng intdef

File: app/src/main/java/com/example/admin/soundcloud_13/screen/home/HomeActivity.java [hunk 9]
```diff
@@ -1,15 +1,29 @@
...
 
-public class HomeActivity extends AppCompatActivity {
-
+public class HomeActivity extends BaseActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
+        TabLayout tabLayout = findViewById(R.id.tab_layout_home);
+        final ViewPager viewPager = findViewById(R.id.view_pager_home);
```
Comment (DoanVanToan) [inline app/src/main/java/com/example/admin/soundcloud_13/screen/home/HomeActivity.java]: remove final

**Pull request merged**